### ISSUE TITLE
Layout can‘t adjust for mobile view on Android Firefox Preview

### ISSUE THREAD

**Issue by Myuki** [vuejs/vuepress#2369]
- [x] I confirm that this is an issue rather than a question.

## Bug report

Layout can‘t adjust for mobile view on Android Firefox Preview (org.mozilla.fenix, Version Code: 21082025). This problem occur at v1.4.1. But at v1.4.0 it didn't exist with exactly the same project and config.js.

#### Steps to reproduce

Open website on Android Firefox Preview. I'm sure desktop mode is disable. Unfortunately, I can't reproduce on Firefox (Android) and Chrome (Android).

#### What is expected?

Layout auto adjust for mobile view.

#### What is actually happening?

Layout can‘t adjust for mobile view and keep desktop view.

![Screenshot_20200504-140028_Firefox Preview_Mosaic_14_1_9](https://user-images.githubusercontent.com/27203797/80940415-4cf68000-8e12-11ea-93d5-585ff7148b33.jpg)

#### Other relevant information

- Output of `npx vuepress info` in my VuePress project:

```
Environment Info:

  System:
    OS: Linux 4.4 Debian GNU/Linux bullseye/sid
    CPU: (12) x64 Intel(R) Core(TM) i7-10710U CPU @ 1.10GHz
  Binaries:
    Node: 14.1.0 - /usr/bin/node
    Yarn: 1.22.4 - /usr/bin/yarn
    npm: 6.14.4 - /usr/bin/npm
  Browsers:
    Chrome: Not Found
    Firefox: Not Found
  npmPackages:
    @vuepress/core: Not Found
    @vuepress/theme-default: Not Found
    vuepress: Not Found
  npmGlobalPackages:
    vuepress: Not Found
```

**Comment by Myuki** [comment]
I find it also can be reproduced on Firefox Desktop in Responsive Design Mode. It only occurs in specific status.

**Comment by Mister-Hope** [comment]
I belive this error belongs to firefox bro.

Vuepress is using `@media () {}` in css for different layouts, MDN shows that it's support by fixfox (obviously, too).

**Comment by takenspc** [comment]
It seems that vuepress remove `<meta name="viewport" content="width=device-width,initial-scale=1">` in client JavaScript.
https://github.com/vuejs/vuepress/blob/master/packages/@vuepress/core/lib/client/root-mixins/updateMeta.js
I think preserving `<meta name="viewport" content="width=device-width,initial-scale=1">` will fix this issue.

**Comment by Mister-Hope** [comment]
> It seems that vuepress remove `<meta name="viewport" content="width=device-width,initial-scale=1">` in client JavaScript.
> https://github.com/vuejs/vuepress/blob/master/packages/@vuepress/core/lib/client/root-mixins/updateMeta.js
> I think preserving `<meta name="viewport" content="width=device-width,initial-scale=1">` will fix this issue.

`<meta name="viewport" content="width=device-width,initial-scale=1">` should be the browser's default behaviour.

**Comment by takenspc** [comment]
> `<meta name="viewport" content="width=device-width,initial-scale=1">` should be the browser's default behaviour.

I totally agree that `<meta name="viewport" content="width=device-width, initial-scale=1">` SHOULD be the default though it isn't. If it was true, we don’t need specify `<meta name="viewport" content="width=device-width, initial-scale=1">` in HTML templates ([index.dev.html](https://github.com/vuejs/vuepress/blob/master/packages/%40vuepress/core/lib/client/index.dev.html) and [index.ssr.html](https://github.com/vuejs/vuepress/blob/master/packages/%40vuepress/core/lib/client/index.ssr.html) ) at all but we do.

Though I'm not a big fun of boilerplates such as `<!DOCTYPE html>`, `<meta charset="utf-8">`, `<meta name="viewport" content="width=device-width, initial-scale=1">`, we can't omit them.

https://hsivonen.fi/utf-8-detection/

> In this regard, `<meta charset=utf-8>` is just like `<!DOCTYPE html>` and `<meta name="viewport" content="width=device-width, initial-scale=1">`. Everyone wants newly-authored content to use UTF-8, the No-Quirks Mode (better known as the Standards Mode), and to work well on small screens. Yet, every single newly-authored HTML document has to explicitly opt in to all three, since it isn’t realistic to get all legacy pages to opt out.

While I understand that it is far from ideal, please preserve boilerplates in production content.

**Comment by Sixteenleft** [comment]
Sorry but I'm new to Vuepress and I'm a bit confused here. How can we prevent this behavior? Should I add my own HTML templates?

**Comment by kbrosnan** [comment]
Chrome behavior here is not spec compliant. Removing the meta viewport tag should display the desktop site.

* https://bugs.chromium.org/p/chromium/issues/detail?id=986674
* https://github.com/webcompat/web-bugs/issues/52296
* https://bugzilla.mozilla.org/show_bug.cgi?id=1650286

**Comment by ksy36** [comment]
This issue was originally reported in https://github.com/vuejs/vuepress/issues/2332 and there is a PR fixing it https://github.com/vuejs/vuepress/pull/2333, however the PR wasn't merged yet. 
Maybe @sakihet or @ludanxer could give some context on this

**Comment by NeverBehave** [comment]
Just a quick mention here, if you wanna have a workaround for now without touching any code in Vuepress, simply add a
 line in the head section of `conflg.js` to fix it.
e.g. https://github.com/DIYgod/RSSHub/blob/e0d4d93d3305c5b806715444c3300a1f37fc50d5/docs/.vuepress/config.js#L61
